internal/core/key.go

```diff
@@ -626,6 +626,25 @@ func (k *IndexDataStoreKey) ToString() string {
 	return string(k.Bytes())
 }
 
+// IsEqual returns true if the two keys are equal
+func (k *IndexDataStoreKey) IsEqual(other IndexDataStoreKey) bool {
+	if k.CollectionID != other.CollectionID || k.IndexID != other.IndexID {
+		return false
+	}
+
+	if len(k.Fields) != len(other.Fields) {
+		return false
```

Comment:
suggestion: Might be worth hitting if not too much hassle.

Reply:
done